src/main/java/model/GameModel.java

```diff
@@ -8,38 +8,69 @@
 import database.DatabaseConnector;
 
 public class GameModel extends Observable<GameObserver> {
+
+	private final DatabaseConnector connector;
+
+	private Question[] questionlist;
+	private int currentquestion;
 	
-    private final DatabaseConnector connector;
-    
-    private Question[] questionlist;
-    
-
-    public GameModel() {
-        this.connector = new DatabaseConnector();
-        Question[] all = connector.getAllQuestions();
-        questionlist = new Question[all.length];
-        
-        
-        //Shuffle the Questions
-        int[] values = new int[all.length];
-        Random r = new Random();
-        for (int i = 0; i < all.length; i++) {
-            values[i] = r.nextInt(10000);
-        }
-
-        for (int i = 0; i < all.length; i++) {
-            int smallest = values[0];
-            int index = 0;
-            for (int j = 1; j < values.length; j++) {
-                if (values[j] < smallest) {
-                    smallest = values[j];
-                    index = j;
-                }
-            }
-            values[index] = Integer.MAX_VALUE;
-            questionlist[i] = all[index];
-        }
-        //Shuffle complete.
-        
-    }
+	private int score;
+	private final int POINTS_PER_QUESTION = 1;
+
+	public GameModel() {
+		this.connector = new DatabaseConnector();
+		currentquestion = -1;
+		score = 0;
+		Question[] all = connector.getAllQuestions();
+		questionlist = new Question[all.length];
+
+		// Shuffle the Questions
+		int[] values = new int[all.length];
+		Random r = new Random();
+		for (int i = 0; i < all.length; i++) {
+			values[i] = r.nextInt(10000);
+		}
+
+		for (int i = 0; i < all.length; i++) {
+			int smallest = values[0];
+			int index = 0;
+			for (int j = 1; j < values.length; j++) {
+				if (values[j] < smallest) {
+					smallest = values[j];
+					index = j;
+				}
+			}
+			values[index] = Integer.MAX_VALUE;
+			questionlist[i] = all[index];
+		}
+		// Shuffle complete.
+
+	}
+
+	/**
+	 * @return The next question in the list or null,
+	 * if there are no more questions available
+	 */
+	public Question getNextQuestion() {
+		if (++currentquestion >= questionlist.length) {
```

Comment:
Nit: `++Syntax` in expressions is not very concise; please consider extracting it as a statement.